src/gflip/gflip_engine.cpp

```diff
@@ -88,6 +136,35 @@ void gflip_engine:: reformulate_to_bagofdistances(void)
 
 // ---------------------------------------------------------
 
+
+double gflip_engine::norm_gfp(scan_bow & bow)
+{
+    double& norm_wgv = bow.norm_wgv;
+    norm_wgv = 0;
+    int s;
+    double combo, sum_idf;
+    std::map<int,std::vector<int> >::const_iterator iter = bow.normgfp_rc.begin();
+
+    for(; iter != bow.normgfp_rc.end(); ++iter)
+     {
+       const std::vector<int>& v = iter->second;
+       s = v.size();
+       if(s<wgv_kernel_size)
+         continue;
+       sum_idf = 0;
+       combo = cached_binomial_coeff[s - 1];
+       for(int j = 0; j<s; ++j)
```

Comment:
optimization: if `cachec_binomial_coeff` size is fix and this code is called many times, we can iterate with `int16` instead of `int`